src/Core/Cache/StandardCacheWarmer.php

```diff
@@ -0,0 +1,297 @@
+<?php
+namespace TYPO3Fluid\Fluid\Core\Cache;
+
+/*
+ * This file belongs to the package "TYPO3 Fluid".
+ * See LICENSE.txt that was shipped with this package.
+ */
+
+use TYPO3Fluid\Fluid\Core\Compiler\FailedCompilingState;
+use TYPO3Fluid\Fluid\Core\Compiler\StopCompilingException;
+use TYPO3Fluid\Fluid\Core\Parser\ParsedTemplateInterface;
+use TYPO3Fluid\Fluid\Core\Parser\SyntaxTree\Expression\ExpressionException;
+use TYPO3Fluid\Fluid\Core\Parser\TemplateParser;
+use TYPO3Fluid\Fluid\Core\Rendering\RenderingContextInterface;
+use TYPO3Fluid\Fluid\Core\ViewHelper\Exception\NotFoundException;
+use TYPO3Fluid\Fluid\View\Exception;
+use TYPO3Fluid\Fluid\View\TemplatePaths;
+
+/**
+ * Class StandardCacheWarmer
+ *
+ * Responsible for performing a full warmup process.
+ * Receives just the RenderingContext (which can be custom for the
+ * framework that invokes the warmup) and resolves all possible
+ * template files in all supported formats and triggers compiling
+ * of those templates.
+ *
+ * The compiling process can be supported in detail in templates
+ * directly through using the `f:cache.*` collection of ViewHelpers.
+ * The compiler is put into a special warmup mode which can in turn
+ * be checked by ViewHelpers when compiling which allows third-party
+ * ViewHelpers to more closely control how they are compiled, if
+ * they are at all compilable.
+ *
+ * The result of the warmup process is returned as a
+ * FluidCacheWarmupResult instance with reports for every template
+ * file that was detected duringthe process; detailing whether or
+ * not the template file was compiled, some metadata about the
+ * template such as which Layout it uses, if any, and finally adds
+ * mitigation suggestions when a template cannot be compiled.
+ *
+ * The mitigation suggestions are specifically generated by this
+ * class and can be elaborated or changed completely by any third-
+ * party implementation of FluidCacheWarmerInterface which allows
+ * them to be specific to the framework in which Fluid is used.
+ * The default set of mitigation suggestions are based on the
+ * standard errors which can be thrown by the Fluid engine.
+ */
+class StandardCacheWarmer implements FluidCacheWarmerInterface {
+
+	/**
+	 * Warm up an entire collection of templates based on the
+	 * provided RenderingContext (the TemplatePaths carried by
+	 * the RenderingContext, to be precise).
+	 *
+	 * Returns a FluidCacheWarmupResult with result information
+	 * about all detected template files and the compiling of
+	 * those files. If a template fails to compile or throws an
+	 * error, a mitigation suggestion is included for that file.
+	 *
+	 * @param RenderingContextInterface $renderingContext
+	 * @return FluidCacheWarmupResult
+	 */
+	public function warm(RenderingContextInterface $renderingContext) {
+		$renderingContext->getTemplateCompiler()->enterWarmupMode();
+		$formats = array('html', 'xml', 'txt', 'json', 'rtf', 'atom', 'rss');
```

Comment:
does it make sense, to make this a protected property, so you could extend and override this easily?